src/App.js

```diff
@@ -89,14 +99,48 @@ function App() {
     setCardsText([...newCards]);
   };
 
+  const addToDeleteListHandler = (cardId, isSelected, isEdited) => {
+    //adding selected cards to the list
+    if (isSelected && !isEdited) {
+      setDeleteList((prevList) => [...prevList, cardId]);
+    //deleting cards from the list if they were unselected
+    } else if ((!isSelected && !isEdited) || (isEdited && !isSelected)) {
+      let index = deleteList.indexOf(cardId);
+
+      if (index !== -1) {
+        let tempList = [...deleteList];
+        tempList.splice(index, 1);
+        setDeleteList(() => {
+          return [...tempList];
+        });
```

Comment:
For such cases you can use shorter version of arrow function syntax:
` setDeleteList(() => ([...tempList]));`